pkg/trace/agent/agent.go

```diff
@@ -68,8 +68,8 @@ type Agent struct {
 	// DiscardSpan will be called on all spans, if non-nil. If it returns true, the span will be deleted before processing.
 	DiscardSpan func(*pb.Span) bool
 
-	// ModifySpan will be called on all spans, if non-nil.
-	ModifySpan func(*pb.Span)
+	// ModifySpan will be called on all spans, if non-nil, along with the corresponding trace chunk.
```

Comment:
Just to make it as clear as possible for future developers who will use the API: Let's document that the modifier function is called in a loop and that the modifications introduced to `*pb.TraceChunk` in iteration N can be overwritten in iteration N+1!

Reply:
Done in ed3bac00dd